src/routes/home/components/DisciplineCard.tsx

```diff
@@ -63,25 +68,43 @@ const DisciplineCard = ({
   }
 
   if (!discipline) {
-    return (
-      <Card>
-        {error?.message === ForbiddenError && isTypeLoadProtected(identifier) ? (
-          <>
-            <ErrorMessageForbidden>
-              {getLabels().home.errorLoadCustomDiscipline} {identifier.apiKey}
-            </ErrorMessageForbidden>
-            <ButtonContainer>
-              <Button
-                onPress={() => AsyncStorage.removeCustomDiscipline(identifier.apiKey).then(refreshHome)}
-                label={getLabels().home.deleteProfession}
-                buttonTheme={BUTTONS_THEME.outlined}
-              />
-            </ButtonContainer>
-          </>
-        ) : (
+    if (error?.message === NetworkError) {
+      return (
+        <Card>
           <ErrorMessage error={error} refresh={refresh} contained />
-        )}
-      </Card>
+        </Card>
+      )
+    }
+
+    let deleteItem
+    let errorMessage
+    if (error?.message === ForbiddenError && isTypeLoadProtected(identifier)) {
+      deleteItem = () => AsyncStorage.removeCustomDiscipline(identifier.apiKey).then(refreshHome)
+      errorMessage = `${getLabels().home.errorLoadCustomDiscipline} ${identifier.apiKey}`
+    } else {
+      deleteItem =
+        'disciplineId' in identifier
```

Comment:
You can just use `!isTypeLoadProtected(identifier)` here, can't you?

Reply:
I'm not sure quite why, but reordering the conditions to use `!isTypeLoadProtected(identifier)` fails the test for unknown errors.

More importantly though, I'd argue that unknown error should be the "fallback" if no other error matches, so explicitly testing if the error is a ForbiddenError makes sense imo.

Reply:
Or maybe I am misunderstanding, do you want to use `!isTypeLoadProtected(identifier)` on line 84 of your quotation, or as a replacement for 85/86?

Reply:
I thought as a replacement for ` 'disciplineId' in identifier` check.

Reply:
All right, that appears to work! Pushed the change.